pointers/pointers.c

```diff
@@ -28,30 +35,60 @@ void string_copy(char *x, char *y)
 */
 char *find_char(char *str, int c)
 {
+  /* char target; */
+  char *target;
+  /* for(char i = *str; i != '\0'; *str++){ */
+  /*   if(i == c) */
+  /*   { */
+  /*     target = i; */
+  /*   } */
 
+  /* } */
+  while(*str != '\0')
+  {
+    /* check the value at address str */
+    /* check it for the target value which passed in and represented by c */
+
+    if(*str == c)
+    {
+      /* Here we take the target variable and set it to the address of the charachter we want*/
+      target = str;
+    }
+  /* increment the address to the next value in the string */
+  str++;
```

Comment:
muuuch better looking. again, it also works with the `*` before the `str++`  (line 58)